.github/workflows/docker.yaml

```diff
@@ -0,0 +1,52 @@
+# https://docs.github.com/en/actions/publishing-packages/publishing-docker-images
+#
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+# GitHub recommends pinning actions to a commit SHA.
+# To get a newer version, you will need to update the SHA.
+# You can also reference a tag or branch, but the action may change without warning.
+
+name: Create and publish a Docker image
+
+on:
+  push:
+    branches: ['release']
```

Comment:
should urn it on this branch to see if it works

Reply:
also should just run it on master

Reply:
i don't think github secrets work on PR's

Reply:
they do

Reply:
i'd rather it kept to releases, these docker builds might take awhile and it's not doing much in the ways of caching

Reply:
it tried to push but it got denied because the write permission shouldn't be active on PR's 

```
Error: buildx call failed with: ERROR: denied: permission_denied: write_package
```